sfcasts/ep1/hungry-dinos.md

```diff
@@ -1,75 +1,78 @@
-let's go back and into dinosaur class so we can fix our accepting visitors method.
-And we'll scroll down to is accepting visitors and ha there's the problem. This
-health is identical to health status healthy. Mm let's. Change this to this health
-status is not identical to health status sick. Then we return true. Otherwise we
-return false. Alright, let's move back into our browser and vendor been PhD unit.
-We're going to do our filter plug again, but aside from, uh, passing in test method,
-names or test classes, we can also do hungry. I know is accepting visitors. Great. We
-have one test and one assertion. This was the test that was failing. As you can see,
-we fixed two problems. We can, we fixed the two problems, one our browsers up and
-running. Again, two, we have passing tests and three we've learned how to use the
-filter flag with PHP unit, just to make sure that our tests are back up and running.
-Let's go ahead and run vendor and PHP unit
-
-And great all nine tests. And the 14 assertions are passing..
+# Filtering Out Hungry Dino's
+
+Instead of seeing our dinos on the dashboard, we're seeing a `TypeError` for
+`GithubService`:
+
+> Return value must be of type `HealthStatus`, `null` returned
+
+Which isn't doing a good job of telling us *what* the problem really is. Thanks
+to the stack trace provided by Symfony, this looks like it's being caused by a
+`Status: Hungry` label. On GitHub, Dennis is hungry again after finishing
+his daily exercise routine.
+
+## Our Enum Is Hungry Too
+
+Looking at `HealthStatus`, we don't have a case for hungry dinos. So add
+`case HUNGRY` that returns `Hungry`... then refresh the dashboard.
+
+And... Ya! No more errors...
+
+But, wait... It says that `Dennis` is *not* accepting visitors. He isn't *sick*,
+just *hungry*. GenLab said only sick dino's should not be on exhibit. Besides, who
+doesn't want to see what happens to the goat?
+
+## Test Hungry Dinos Can Have Visitors
+
+In `DinosaurTest`, we need to assert that hungry dino's *can* have visitors.
+Hmm... I think we might be able to use `testIsNotAcceptingVisitorsIfSick()` for this.
+Yup, that's what we'll do. Below, add a `healthStatusProvider()` that returns 
+`\Generator` and for the first dataset `yield 'Sick dino is not accepting visitors'`. 
+In the array say `HealthStatus::SICK`, and `false`. Next, 
+`yield 'Hungry dino is accepting visitors'` with `[HealthStatus::HUNGRY, true]`.
+Above, add the `@dataProvider` annotation so we can use `healthStatusProvider()`.
+While we're here, rename the method to `testIsAcceptingVisitorsBasedOnHealthStatus` 
+then add the arguments `HealthStatus $healthStatus` and `bool $expectedStatus`. 
+Inside set the health with `$healthStatus` then replace `assertFalse()` with 
+`assertSame($expectedStatus)` is identical to `$dino->isAcceptingVisitors()`.
+
+Phew, that was a lot of work.
+
+## Filtering Tests
+
+Let's see if that did the trick. Run:
+
```

Comment:
Is it a faux pas to have a heading right before we run a test? Obviously based on that answer, the "lead in" to the test needs to change.